docs/basics/typescript-integration.md

```diff
@@ -0,0 +1,246 @@
+
+Create the following `codegen.ts` configuration file:
+
+```ts
+import { CodegenConfig } from '@graphql-codegen/cli';
+
+const config: CodegenConfig = {
+  schema: '<YOUR_GRAPHQL_API_URL>',
+  documents: ['src/**/*.tsx'],
+  ignoreNoDocuments: true, // for better experience with the watcher
+  generates: {
+    './src/gql/': {
+      preset: 'client',
+      plugins: [],
+    },
+  },
+};
+
+export default config;
+```
+
+#### Vue project configuration
+
+Create the following `codegen.ts` configuration file:
+
+```ts
+import type { CodegenConfig } from '@graphql-codegen/cli';
+
+const config: CodegenConfig = {
+  schema: '<YOUR_GRAPHQL_API_URL>',
+  documents: ['src/**/*.vue'],
+  ignoreNoDocuments: true, // for better experience with the watcher
+  generates: {
+    './src/gql/': {
+      preset: 'client',
+      config: {
+        useTypeImports: true,
+      },
+      plugins: [],
+    },
+  },
+};
+
+export default config;
+```
+
+#### Svelte project configuration
+
+Create the following `codegen.ts` configuration file:
+
+```ts
+const config: CodegenConfig = {
+  schema: '<YOUR_GRAPHQL_API_URL>',
+  documents: ['src/**/*.svelte'],
+  ignoreNoDocuments: true, // for better experience with the watcher
+  generates: {
+    './src/gql/': {
+      preset: 'client',
+      plugins: [],
+    },
+  },
+};
+
+export default config;
+```
+
+## Typing queries, mutations and subscriptions
+
+Now that your project is properly configured, let's start codegen in watch mode:
+
+```sh
+yarn codegen
+# or
+npm run codegen
+```
+
+This will generate a `./src/gql` folder that exposes a `graphql()` function.
+
+Let's use this `graphql()` function to write our GraphQL Queries, Mutations and Subscriptions.
+
+Here, an example with the React bindings, however, the usage remains the same for Vue and Svelte bindings:
+
+```tsx
+import React from 'react';
+import { useQuery } from '@apollo/client';
+
+import './App.css';
+import Film from './Film';
+import { graphql } from '../src/gql';
+
+const allFilmsWithVariablesQueryDocument = graphql(/* GraphQL */ `
+  query allFilmsWithVariablesQuery($first: Int!) {
+    allFilms(first: $first) {
+      edges {
+        node {
+          ...FilmItem
+        }
+      }
+    }
+  }
+`);
+
+function App() {
+  // `data` is typed!
+  const { data } = useQuery(allFilmsWithVariablesQueryDocument, { variables: { first: 10 } });
+  return (
+    <div className="App">
+      {data && (
+        <ul>
+          {data.allFilms?.edges?.map(
+            (e, i) => e?.node && <Film film={e?.node} key={`film-${i}`} />
+          )}
+        </ul>
+      )}
+    </div>
+  );
+}
+
+export default App;
+```
+
+_Examples with Vue are available [in the GraphQL Code Generator repository](https://github.com/dotansimha/graphql-code-generator/tree/master/examples/front-end/vue/urql)_.
+
+Using the generated `graphql()` function to write your GraphQL document results in instantly typed result and variables for queries, mutations and subscriptions!
```

Comment:
maybe for `graphql()` we should link to codgen docs where we explain in depth what this does vs just tagged literal
```suggestion
Using the generated `graphql()` function to write your GraphQL document results in instantly typed result and variables for queries, mutations and subscriptions!
```

Reply:
I'm not sure we need to explain how it works under the hood in the docs 👀 
Why not in a blog post?